llvm/lib/Transforms/InstCombine/InstCombineLoadStoreAlloca.cpp

```diff
@@ -764,6 +773,7 @@ static Instruction *unpackLoadToAggregate(InstCombinerImpl &IC, LoadInst &LI) {
           Name + ".unpack");
       // Propagate AA metadata. It'll still be valid on the narrowed load.
       L->setAAMetadata(LI.getAAMetadata());
```

Comment:
The AATag should be adjusted to use the new offset and size. cc @nikic

Reply:
I added a new commit to try to implement this. But it seems to not work for scalable vectors (llvm/test/Transforms/InstCombine/scalable-vector-struct.ll) fails to run.

```
LLVM ERROR: Invalid size request on a scalable vector.
```

Reply:
@nikic @dtcxzyw I am not sure how to approach, this should I disable this change if `STRICT_FIXED_SIZE_VECTORS` is not defined?

Reply:
gentle ping!